src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -96,28 +96,30 @@ public List<AttachmentUnit> splitAndSaveUnits(LectureUnitInformationDTO lectureU
     }
 
     /**
-     * Removes the break slides from the given document.
+     * Removes the break slides or solution slides from the given document.
      *
      * @param document document to remove break slides from
      */
-    private void removeBreakSlides(PDDocument document) {
+    private void removeBreakOrSolutionSlides(PDDocument document, boolean removeBreakSlides, boolean removeSolutionSlides) {
 
         try {
             PDFTextStripper pdfTextStripper = new PDFTextStripper();
             Splitter pdfSplitter = new Splitter();
             List<PDDocument> pages = pdfSplitter.split(document);
-            Iterator<PDDocument> iterator = pages.listIterator();
 
-            int index = 0;
-            while (iterator.hasNext()) {
-                PDDocument currentPage = iterator.next();
+            // Uses a decrementing loop (starting from the last index) to ensure that the
+            // index values are adjusted correctly when removing pages.
+            for (int index = pages.size() - 1; index >= 0; index--) {
+                PDDocument currentPage = pages.get(index);
                 String slideText = pdfTextStripper.getText(currentPage);
-                if (isBreakSlide(slideText)) {
+
+                if (isBreakSlide(slideText) && removeBreakSlides) {
```

Comment:
You could put the removeBreakSlides conditions in front, to avoid unnecessary calls to isBreakSlide if it’s disabled anyway. But it is not super important in this case.
Same for solution slides.

Reply:
I think in this case it will not make any difference in terms of performance because `isBreakSlide` is only checking if it contains the string and moreover this function is only called for Break and Example solution slides.